administrator/components/com_content/controllers/article.php

```diff
@@ -154,8 +155,31 @@ public function batch($model = null)
 	 *
 	 * @since	3.1
 	 */
-	protected function postSaveHook(JModelLegacy $model, $validData = array())
+	protected function postSaveHook(JModelLegacy $model, $validData)
 	{
+		JModelLegacy::addIncludePath(JPATH_ADMINISTRATOR . '/components/com_menus/models', 'MenusModel');
+		$itemModel = JModelAdmin::getInstance('Item', 'MenusModel');
+		$itemModel->addTablePath(JPATH_ADMINISTRATOR . '/components/com_menus/tables');
+		$articleId = $model->getState($this->context . '.id')
+
+		$menuData['id'] = 0;
+		$menuData['menutype'] = $validData['menulink']['menutype'];
+		$menuData['title'] = $validData['menulink']['menutitle'];
```

Comment:
@shubhamnba2009 you can also create the array the following way:

$menuData = array(
	'id' => JComponentHelper::getComponent('com_content')->id,
	'menutype' => ...
);

So you don't have to repeat $menuData so often and easier see the key. But it's a bit depending on content.